keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -66,19 +70,19 @@ File and block are first class citizens within the Kubernetes ecosystem.  Object
 + Define a native _data-plane_ object store API which would greatly improve object store app portability.
 
 ## Vocabulary
-+  _Brownfield_ - also called "dynamic brownfield" - buckets are created outside the COSI system but granted access via COSI. The OB, driver secret, app secret and binding are all performed by COSI.
-+ _Bucket_ - A namespace where objects reside, similar to a flat POSIX directory.
-+ _BucketClass_ (BC) - A cluster-wide (non-namespaced) custom resource containing fields defining the provisioner and an immutable parameter set.  Referenced by ObjectBucketClaims and populated into the OB.  Only used for provisioning new buckets.
-+ _Container Object Storage Interface (COSI)_ -  The specification of gRPC data and methods making up the communication protocol between the driver and the sidecar.
-+ _COSI Controller_ - A single, central controller which manages OBCs, OBs, and Secrets cluster-wide. Often referred to below as the "central controller".
-+ _"cosi-system"_ - The namespace name for all COSI drivers and sidecars. Even if a cluster supports different, concurrent object stores, all drivers live in this namespace.
-+ _Driver - A containerized gRPC server which implements a storage vendor’s business logic through the COSI interface. It can be written in any language supported by gRPC and is independent of Kubernetes. There is historical precedence in Kubernetes to call this type of container a _plugin_, but going forward _driver_ is the preferred name.
-+ _Greenfield_ - new buckets dynamically created by the driver.
-+ _OB_ (Object Bucket) - A cluster-wide (non-namespaced) custom resource representing the provisioned bucket and relevant metadata.
-+ _OBC_ (Object Bucket Claim) - A user-namespaced custom resource representing a user’s bucket request. 
-+ _Object_ - An atomic, immutable unit of data stored in buckets.
-+ _Sidecar_ - A controller deployed in the same pod as the driver, responsible for managing OBs and communicating with the Driver via gRPC. Needs write access to its OB. Note: the sidecar controller can use [predicates](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/predicate) to filter OBs by driver name (assuming a unique label is added to OBs).
-+ _Static_ - also called "static brownfield" - buckets are created outside the COSI system but granted access via COSI without the need for the driver/sidecar. The OB and driver secret are created manually in the expected locations. The app secret and binding are performed by COSI.
++  _Brownfield Bucket_ - externally created, represented by a BuckteClass and managed by a provisioner
++ _Bucket_ - A user-namespaced custom resource representing an object store bucket.
```

Comment:
The problem I see with this name is that it's hard to know, when using the word _bucket_ if one is referring to a real object store's bucket or the k8s abstraction of a bucket request?

Reply:
Persistent comes with the same potential for confusion that Claim does.  I don't think we necessarily need to worry about the shorthand versions.  plenty of APIs don't include them at all.  Rook-ceph for instance.